### PR TITLE
Prepare Release v1.30.0

### PR DESCRIPTION
**Pull request by cmslewis**
:book: **Latest docs**: [blueprintjs.com/docs](http://blueprintjs.com/docs)

### 🔄 Changed
- #1604 Moved the following `Utils` functions from `packages/table` into `packages/core`:
    - `arraysEqual`
    - `deepCompareKeys`
    - `getDeepUnequalKeyValues`
    - `getShallowUnequalKeyValues`
    - `shallowCompareKeys`

### 🐛 Bug fixes
- #1608 `EditableText` now elegantly handles long strings with no spaces. (🎩 @luyao-wang)
image
- #1616 `InputGroup` now properly styles `pt-spinner`s contained in child buttons.
image

### 🔄 Changed
- #1627 `Table` now supports the `useInteractionBar` prop.
    - `ColumnHeaderCell`'s `useInteractionBar` prop is now deprecated.
    - We made this change for two reasons:
        - From a design perspective, the interaction bar should be enabled for all cells at once or for none at all. Thus, it makes sense to provide the flag to the top-level `Table` component.
        - This change enables a fix for a bug where the column header wouldn't resize properly when `useInteractionBar` was toggled on and off.
- #1604 Moved the following `Utils` functions from `packages/table` into `packages/core`:
    - `arraysEqual`
    - `deepCompareKeys`
    - `getDeepUnequalKeyValues`
    - `getShallowUnequalKeyValues`
    - `shallowCompareKeys`
- #1596 #1621 `Table` now renders and updates fewer DOM elements under the hood in certain cases, leading to slightly faster rendering performance.

### 🐛 Bug fixes
- #1621 `Table` now aligns the left selection/focused-cell border properly when row headers are hidden:
image
- #1622 `Table` now shows a left border on selected regions that include cells in the first column when row headers are visible. 
- #1624 `TruncatedFormat` now does fewer DOM reads when the content popover is closed, leading to slightly improved rendering performance.

### 💎 New features
- #1581 `Popover2` now offers shims for renamed props to ease migration from `Popover` (🎩 @brieb)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "blueprint-monorepo",
-  "version": "1.28.0",
```
Comment (cmslewis) [inline package.json]: Oops, we didn't bump this for 1.29.0.

File: packages/table/package.json [hunk 2]
```diff
@@ -20,7 +20,7 @@
     "watch": "onchange 'src/**' 'preview/*.ts*' -- npm-run-all build:gulp build:preview"
   },
   "dependencies": {
-    "@blueprintjs/core": "^1.25.0",
+    "@blueprintjs/core": "^1.30.0",
```
Comment (cmslewis) [inline packages/table/package.json]: Bumping because of the utils that moved from `table` to `core`.

**Comment by adidahiya** [comment]
also please remove tslint changes from the release notes.

**Comment by cmslewis** [comment]
@adidahiya @giladgray a lot of the removals seem to be related to `moment-timezone`. I'm seeing an error related to it when the `prepRelease.sh` script runs:

```tsx
[23:24:17] Finished 'bundle-core' after 1.5 s
[23:24:17] [webpack] Time: 1580ms
             Asset    Size  Chunks             Chunk Names
    labs.bundle.js  279 kB       0  [emitted]  labs
labs.bundle.js.map  331 kB       0  [emitted]  labs

ERROR in ./packages/labs/~/moment-timezone/data/packed/latest.json
Module parse failed: /Users/clewis/Developer/blueprint-external/packages/labs/node_modules/moment-timezone/data/packed/latest.json Unexpected token (2:10)
You may need an appropriate loader to handle this file type.
SyntaxError: Unexpected token (2:10)
  at Parser.pp$4.raise (/Users/clewis/Developer/blueprint-external/node_modules/webpack/node_modules/acorn/dist/acorn.js:2221:15)
  at Parser.pp.unexpected (/Users/clewis/Developer/blueprint-external/node_modules/webpack/node_modules/acorn/dist/acorn.js:603:10)
  at Parser.pp.semicolon (/Users/clewis/Developer/blueprint-external/node_modules/webpack/node_modules/acorn/dist/acorn.js:581:61)
  at Parser.pp$1.parseExpressionStatement (/Users/clewis/Developer/blueprint-external/node_modules/webpack/node_modules/acorn/dist/acorn.js:966:10)
  at Parser.pp$1.parseStatement (/Users/clewis/Developer/blueprint-external/node_modules/webpack/node_modules/acorn/dist/acorn.js:730:24)
  at Parser.pp$1.parseBlock (/Users/clewis/Developer/blueprint-external/node_modules/webpack/node_modules/acorn/dist/acorn.js:981:25)
  at Parser.pp$1.parseStatement (/Users/clewis/Developer/blueprint-external/node_modules/webpack/node_modules/acorn/dist/acorn.js:709:33)
  at Parser.pp$1.parseTopLevel (/Users/clewis/Developer/blueprint-external/node_modules/webpack/node_modules/acorn/dist/acorn.js:638:25)
  at Parser.parse (/Users/clewis/Developer/blueprint-external/node_modules/webpack/node_modules/acorn/dist/acorn.js:516:17)
  at Object.parse (/Users/clewis/Developer/blueprint-external/node_modules/webpack/node_modules/acorn/dist/acorn.js:3098:39)
  at Parser.parse (/Users/clewis/Developer/blueprint-external/node_modules/webpack/lib/Parser.js:902:15)
  at NormalModule.<anonymous> (/Users/clewis/Developer/blueprint-external/node_modules/webpack/lib/NormalModule.js:104:16)
  at NormalModule.onModuleBuild (/Users/clewis/Developer/blueprint-external/node_modules/webpack-core/lib/NormalModuleMixin.js:310:10)
  at nextLoader (/Users/clewis/Developer/blueprint-external/node_modules/webpack-core/lib/NormalModuleMixin.js:275:25)
  at /Users/clewis/Developer/blueprint-external/node_modules/webpack-core/lib/NormalModuleMixin.js:259:5
  at Storage.finished (/Users/clewis/Developer/blueprint-external/node_modules/webpack/node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:38:16)
  at /Users/clewis/Developer/blueprint-external/node_modules/graceful-fs/graceful-fs.js:78:16
  at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:504:3)

 @ ./packages/labs/~/moment-timezone/index.js 2:15-51
[23:24:17] Finished 'bundle-labs' after 1.58 s
```

I'm also seeing the following later in the build:

```tsx

[23:24:36] [webpack] Time: 16627ms
           Asset     Size  Chunks             Chunk Names
    site-docs.js  5.39 MB       0  [emitted]  site-docs
site-docs.js.map  6.63 MB       0  [emitted]  site-docs

ERROR in ./packages/labs/examples/omniboxExample.tsx
(17,5): error TS6133: 'IToastProps' is declared but never used.

ERROR in ./packages/labs/src/components/timezone-picker/timezoneMetadata.ts
(30,26): error TS2339: Property 'population' does not exist on type 'MomentZone'.

ERROR in /Users/clewis/Developer/blueprint-external/packages/labs/node_modules/popper.js/index.d.ts
(9,15): error TS2300: Duplicate identifier 'Position'.

ERROR in /Users/clewis/Developer/blueprint-external/packages/labs/node_modules/popper.js/index.d.ts
(11,15): error TS2300: Duplicate identifier 'Placement'.

ERROR in /Users/clewis/Developer/blueprint-external/packages/labs/node_modules/popper.js/index.d.ts
(27,15): error TS2300: Duplicate identifier 'Boundary'.

ERROR in /Users/clewis/Developer/blueprint-external/packages/labs/node_modules/popper.js/index.d.ts
(29,15): error TS2300: Duplicate identifier 'ModifierFn'.

ERROR in /Users/clewis/Developer/blueprint-external/packages/labs/node_modules/popper.js/index.d.ts
(62,5): error TS2374: Duplicate string index signature.

ERROR in /Users/clewis/Developer/blueprint-external/packages/labs/node_modules/popper.js/index.d.ts
(106,24): error TS2300: Duplicate identifier 'Popper'.

ERROR in /Users/clewis/Developer/blueprint-external/packages/site-docs/node_modules/@blueprintjs/labs/node_modules/popper.js/index.d.ts
(9,15): error TS2300: Duplicate identifier 'Position'.

ERROR in /Users/clewis/Developer/blueprint-external/packages/site-docs/node_modules/@blueprintjs/labs/node_modules/popper.js/index.d.ts
(11,15): error TS2300: Duplicate identifier 'Placement'.

ERROR in /Users/clewis/Developer/blueprint-external/packages/site-docs/node_modules/@blueprintjs/labs/node_modules/popper.js/index.d.ts
(27,15): error TS2300: Duplicate identifier 'Boundary'.

ERROR in /Users/clewis/Developer/blueprint-external/packages/site-docs/node_modules/@blueprintjs/labs/node_modules/popper.js/index.d.ts
(29,15): error TS2300: Duplicate identifier 'ModifierFn'.

ERROR in /Users/clewis/Developer/blueprint-external/packages/site-docs/node_modules/@blueprintjs/labs/node_modules/popper.js/index.d.ts
(106,24): error TS2300: Duplicate identifier 'Popper'.
[23:24:36] Finished 'webpack-docs' after 17 s
[23:24:36] Finished 'build' after 34 s

```

**Could any of these errors be causing undesired deletions from the `site-docs` bundle**?

File: packages/table/package.json (hunk 2 above)
Comment (giladgray) [reply packages/table/package.json]: is this the correct minimum version?

Comment (llorca) [reply packages/table/package.json]: Yup, as Chris mentioned, some utils moved from `table` to `core`

**Comment by cmslewis** [comment]
@giladgray is taking a look at this release today to see if things look good.

File: packages/table/package.json (hunk 2 above)
Comment (cmslewis) [reply packages/table/package.json]: Yes

**Comment by giladgray** [comment]
ok pushed some build fixes. we should still see the `timezoneMetadata` and `popper.js` errors noted at the end of https://github.com/palantir/blueprint/pull/1642#issuecomment-333357008 (until #1527 is resolved), but the `labs.bundle.js` error should be fixed.

**Comment by giladgray** [comment]
@adidahiya @cmslewis pretty sure the -40k lines on docs.js is entirely due to `prettier` formatting things to require fewer lines. for instance, the massive `docs.json` blob now sits on exactly **one** line, instead of **thirty-two thousand**.

docs-line-diff

File: gulp/util/webpack-config.js [hunk 3]
```diff
@@ -13,6 +13,12 @@ const globalName = (id) => upperFirst(camelCase(id));
 
 const DEFAULT_CONFIG = {
     devtool: "source-map",
+    module: {
+        loaders: [
+            // always provide JSON loader to support importing data files (see moment-timezone)
+            { loader: "json-loader", test: /\.json$/ },
```
Comment (giladgray) [inline gulp/util/webpack-config.js]: this resolve the `latest.json` issue from `labs.bundle.js`

File: packages/table/package.json (hunk 2 above)
Comment (giladgray) [reply packages/table/package.json]: well sure but didn't that actually happen _last_ release? 1.29.0?

no big deal though.

**Comment by adidahiya** [comment]
@giladgray @cmslewis `prettier` shouldn't be running on built files; did you run `gulp lint-fix` in this PR?

File: packages/table/package.json (hunk 2 above)
Comment (giladgray) [reply packages/table/package.json]: nvm, I see release notes. my bad.

**Comment by cmslewis** [comment]
@adidahiya no, I didn't.

**Comment by DrewDennison** [comment]
@cmslewis did this make it to NPM yet?

**Comment by adidahiya** [comment]
The release build failed; we're looking into it now.

**Pull request merged**